storagenode/storagenodedb/database.go

```diff
@@ -232,22 +205,35 @@ func (db *DB) Close() error {
 		db.ndb.Close(),
 		db.adb.Close(),
 
-		db.versionsDB.Close(),
-		db.v0PieceInfoDB.Close(),
-		db.bandwidthDB.Close(),
-		db.ordersDB.Close(),
-		db.pieceExpirationDB.Close(),
-		db.pieceSpaceUsedDB.Close(),
-		db.reputationDB.Close(),
-		db.storageUsageDB.Close(),
-		db.usedSerialsDB.Close(),
-		db.satellitesDB.Close(),
+		db.closeDatabases(),
 	)
 }
 
-// Versions returns the instance of the versions database.
-func (db *DB) Versions() SQLDB {
-	return db.versionsDB
+// closeDatabases closes all the SQLite database connections and removes them from the associated maps.
+func (db *DB) closeDatabases() error {
+	var err error
+
+	for k := range db.sqlDatabases {
+		_ = errs.Combine(err, db.closeDatabase(k))
+	}
+	return err
+}
+
+// closeDatabase closes the specified SQLite database connections and removes them from the associated maps.
+func (db *DB) closeDatabase(dbName string) (err error) {
```

Comment:
Why handle them by name rather than by directly calling the method on the specific database implementation. It feels like an unnecessary indirection.

Reply:
I introduced this because as this code base is growing the longer the list of databases is becoming. Directly accessing objects in every type of access is starting to become brittle and error prone to copy/paste errors.

Reply:
I'm not quite following. why would `closeDatabase("bandwidth")` is less brittle than `bandwidth.close`. You can also have a map of databases that handle them.

Reply:
If we stick with no array/map/etc list of databases it means every function that needs to operate across all the databases requires copy/pasting code. For example, this wouldn't be possible:

```
func (db *DB) closeDatabases() error {
	var err error

	for k := range db.sqlDatabases {
		err = errs.Combine(err, db.closeDatabase(k))
	}
	return err
}
```

It's brittle in the sense that when someone adds a new database it's very easy to miss spots in the code and bugs are extremely easy to introduce because you should have copy/pasted something you didn't. This alleviates _some_ of that.

I had considered shifting the concern at a higher scoped object (such as database) but this PR is so much redesigning already I felt that was over designing things for what we need right now. There are a lot of different ways to approach OO design and I felt this was already a big improvement over what we had.